From 6cfbcf2f40e371ce36c030addc539597d058b3a9 Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Wed, 7 Jul 2021 14:55:31 +0900
Subject: [PATCH] ksmbd: remove unneeded NULL check in for_each_netdev
Git-commit: 6cfbcf2f40e371ce36c030addc539597d058b3a9
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


netdev can never be NULL in for_each_netdev loop.
This patch remove unneeded NULL check.

Reported-by: Coverity Scan <scan-admin@coverity.com>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/ksmbd/smb2pdu.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/fs/ksmbd/smb2pdu.c b/fs/ksmbd/smb2pdu.c
index 994b95b6b3c2..2811dfabfa75 100644
--- a/fs/ksmbd/smb2pdu.c
+++ b/fs/ksmbd/smb2pdu.c
@@ -7004,11 +7004,6 @@ static int fsctl_query_iface_info_ioctl(struct ksmbd_conn *conn,
 
 	rtnl_lock();
 	for_each_netdev(&init_net, netdev) {
-		if (unlikely(!netdev)) {
-			rtnl_unlock();
-			return -EINVAL;
-		}
-
 		if (netdev->type == ARPHRD_LOOPBACK)
 			continue;
 
-- 
2.33.0

